File: templates/java-webmvc/subsystem/user/component/src/main/java/cobalt/user/UserRepository.java
package cobalt.user;

import cobalt.user.model.User;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
class UserRepository {

  static List<User> users = List.of(
    new User(1, "Talia", "Fritsch"),
    new User(2, "Daisy", "Lubowitz"),
    new User(3, "Chaz", "Rolfson"),
    new User(4, "Nelda", "Wolff"),
    new User(5, "Giovanna", "Nader"),
    new User(6, "Jerald", "Romaguera"),
    new User(7, "Faustino", "Nienow")
  );

  List<User> findAll() {
    return users;
  }

  Optional<User> findById(Integer id) {
    return users
      .stream()
      .filter(it -> it.id.equals(id))
      .findFirst();
  }

}

File: templates/kotlin-webmvc/subsystem/user/component/src/main/kotlin/{{PACKAGE_NAME}}/user/UserRepository.kt
package {{PACKAGE_NAME}}.user

import {{PACKAGE_NAME}}.user.model.User
import org.springframework.stereotype.Repository

@Repository
internal class UserRepository {

  companion object {
    private val users = listOf(
      User(id = 1, familyName = "Talia", givenName = "Fritsch"),
      User(id = 2, familyName = "Daisy", givenName = "Lubowitz"),
      User(id = 3, familyName = "Chaz", givenName = "Rolfson"),
      User(id = 4, familyName = "Nelda", givenName = "Wolff"),
      User(id = 5, familyName = "Giovanna", givenName = "Nader"),
      User(id = 6, familyName = "Jerald", givenName = "Romaguera"),
      User(id = 7, familyName = "Faustino", givenName = "Nienow")
    )
  }

  fun findAll(): List<User> = users

  fun findById(id: Int): User? = users.find { it.id == id }

}
